src/JitInterface/src/JitConfigProvider.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.JitInterface
+{
+    internal sealed class JitConfigProvider
+    {
+        private Dictionary<string, string> _config = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        private object _keepAlive; // Keeps callback delegates alive
+
+        public IntPtr UnmanagedInstance
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="JitConfigProvider"/>.
+        /// </summary>
+        /// <param name="parameters">Name-value pairs separated by an equals sign.</param>
+        public JitConfigProvider(IEnumerable<string> parameters)
+        {
+            foreach (var param in parameters)
+            {
+                int indexOfEquals = param.IndexOf('=');
+
+                // We're skipping bad parameters without reporting.
+                // This is not a mainstream feature that would need to be friendly.
+                // Besides, to really validate this, we would also need to check that the config name is known.
+                if (indexOfEquals < 1)
+                    continue;
+
+                string name = param.Substring(0, indexOfEquals);
+                string value = param.Substring(indexOfEquals + 1);
+
+                _config[name] = value;
+            }
+
+            UnmanagedInstance = CreateUnmanagedInstance();
+        }
+
+        public int GetIntConfigValue(string name, int defaultValue)
+        {
+            string stringValue;
+            int intValue;
+            if (_config.TryGetValue(name, out stringValue) &&
+                Int32.TryParse(stringValue, out intValue))
```

Comment:
Ok, this flag does not do what one would expect from the name... 

The `COMPlus_XXX` settings are actually always treated as hex values (https://github.com/dotnet/coreclr/blob/32f0f9721afb584b4a14d69135bea7ddc129f755/src/utilcode/regutil.cpp#L239)  so it may avoid some confusion to always parse them as hex numbers here as well.